src/state/editors.ts

```diff
@@ -234,14 +234,27 @@ export class PerEditorState implements vscode.Disposable {
   public notifyDidBecomeActive() {
     const { editor, mode } = this;
 
-    this.extension.statusBar.activeModeSegment.setContent(mode.name);
+    this.extension.statusBar.activeModeSegment.setContent(this._formatDisplayName(mode.name));
 
     editor.options.lineNumbers = mode.lineNumbers;
     editor.options.cursorStyle = mode.cursorStyle;
 
     return vscode.commands.executeCommand("setContext", extensionName + ".mode", mode.name);
   }
 
+  private _formatDisplayName(modeName: string) {
+    switch (vscode.workspace.getConfiguration(extensionName)
+      .get<string>("activeModeDisplayTextTransform")) {
```

Comment:
This is a good way to get the key, but to avoid reading the configuration every time the mode changes (which can occur pretty often), I would instead use `observePreference()` in `src/state/extension.ts` to keep a field of `Extension` in sync with the configuration, and then read from `_extension` in `notifyDidBecomeActive()`.